tracer/src/Datadog.Trace/IAST/IastModule.cs

```diff
@@ -7,33 +7,50 @@
 
 using System;
 using System.Diagnostics;
-using System.Linq;
+using System.Runtime.InteropServices;
 using Datadog.Trace.Configuration;
-using Datadog.Trace.Iast;
-using Datadog.Trace.Util;
 
 namespace Datadog.Trace.Iast
 {
     internal class IastModule
     {
-        private const string OperationNameHash = "weak_hashing";
+        private const string OperationNameWeakHash = "weak_hashing";
+        private const string OperationNameWeakCipher = "weak_cipher";
 
         public IastModule()
         {
         }
 
-        public static Scope? OnHashingAlgorithm(string? algorithm, IntegrationId integrationId, Datadog.Trace.Iast.Iast iast)
+        public static Scope? OnCipherAlgorithm(Type type, IntegrationId integrationId, Iast iast)
+        {
+            var algorithm = type.BaseType?.Name;
+
+            if (algorithm == null || !InvalidCipherAlgorithm(type, algorithm, iast))
```

Comment:
nit: favour `is` to avoid operator overloading
```suggestion
            if (algorithm is null || !InvalidCipherAlgorithm(type, algorithm, iast))
```